File: apps/ios/Barns/BarnsApp/Domain/UseCases/Catalog/GetGreeneryInfoUseCase.swift
/// Returns official basic information for a greenery, if it exists.
struct GetGreeneryInfoUseCase {
    private let repository: GreeneryInfoRepository

    init(repository: GreeneryInfoRepository) {
        self.repository = repository
    }

    func execute(id: GreeneryInfo.ID) async throws -> GreeneryInfo? {
        try await repository.greeneryInfo(id: id)
    }
}

File: apps/android/app/src/main/java/com/barns/app/domain/usecase/catalog/GetGreeneryInfoUseCase.kt
package com.barns.app.domain.usecase.catalog

import com.barns.app.domain.model.GreeneryInfo
import com.barns.app.domain.repository.GreeneryInfoRepository

/** Returns official basic information for a greenery, if it exists. */
class GetGreeneryInfoUseCase(
    private val repository: GreeneryInfoRepository,
) {
    suspend fun execute(id: String): GreeneryInfo? = repository.greeneryInfo(id)
}
